File: 7.整数反转.ts
/*
 * @lc app=leetcode.cn id=7 lang=typescript
 *
 * [7] 整数反转
 */

// @lc code=start
function reverse(x: number): number {
  let tempX: string | number = Math.abs(x);
    tempX = tempX.toString().split("").reverse().join("");
    tempX = Number(tempX);
    if(x<1){
        tempX = tempX * -1;
    }
    if(tempX>Math.pow(2,31) - 1 || tempX <Math.pow(-2,31)){
        tempX = 0;
    }
    return tempX;
};
// @lc code=end

File: 7.整数反转.go
/*
 * @lc app=leetcode.cn id=7 lang=golang
 *
 * [7] 整数反转
 */
package jp_leetcode

import (
	"math"
	"strconv"
	"strings"
)

// @lc code=start
func reverse(x int) int {
	var tempFloat float64 = float64(x)
	var tempInt int = int(math.Abs(tempFloat))
	var tempX string = strconv.Itoa(tempInt)
	var tempArr = strings.Split(tempX, "")
	var abs = math.Pow(2, 31)
	var min = int(abs * -1)
	var max = int(abs - 1)

	for i, j := 0, len(tempArr)-1; i < j; i, j = i+1, j-1 {
		tempArr[i], tempArr[j] = tempArr[j], tempArr[i]
	}

	tempX = strings.Join(tempArr, "")

	tempInt, _ = strconv.Atoi(tempX)

	if x < 1 {
		tempInt = tempInt * -1
	}

	if tempInt < min || tempInt > max {
		tempInt = 0
	}

	return tempInt

}

// @lc code=end

func main() {
	//reverse(123)
	reverse(1534236469)
}
